### ISSUE TITLE
Handles multiple paragraphs and code blocks in list incorrectly

### ISSUE THREAD

**Issue by dominikh** [shurcooL/github_flavored_markdown#8]
Consider the following markdown input:

    - List item 1
    - List item 2
    
      ```
      some code here
      ```
    
      more text here
    
      ```
      more code here
      ```

Your package renders it as

```
<ul>
<li>List item 1</li>
<li>List item 2</li>
</ul>

<pre><code>  some code here
</code></pre>

<p>more text here</p>

<pre><code>  more code here
</code></pre>
```

That is, it closes the list too soon. GitHub renders it correctly: 

- List item 1
- List item 2

  ```
  some code here
  ```

  more text here

  ```
  more code here
  ```

**Comment by shurcooL** [comment]
This is a known behavior of `blackfriday`. Unlike GitHub's markdown renderer, it requires indent to be of size 4 spaces or 1 tab at minimum. GitHub requires a minimum of 2 spaces. See https://github.com/russross/blackfriday/issues/244#issuecomment-194080110.

You should either write (note indent is 4 spaces):

```
- List item 1
- List item 2

    ```
    some code here
    ```

    more text here

    ```
    more code here
    ```
```

Or you [can use tabs](http://virtivia.com:27080/obernhwfgpgn.txt). This is the recommended style and it's what [`markdownfmt`](https://github.com/shurcooL/markdownfmt) uses.

**Comment by dominikh** [comment]
May that as it be, your library has *github* in its name :) – Also, ordinary continuation lines of a list item don't need to be indented deeper than the first line. Why is that not true for additional paragraphs?

**Comment by shurcooL** [comment]
I agree, this is a bug, because this package's intent is to be compatible with GitHub Flavored Markdown rendering as done via their API:

> The functionality should be equivalent to the GitHub Markdown API endpoint specified at https://developer.github.com/v3/markdown/#render-a-markdown-document-in-raw-mode, except the rendering is performed locally.

What I shared above is a workaround you can use until this is resolved.

To resolve this issue, https://github.com/russross/blackfriday/issues/244 needs to be resolved.

**Comment by jlearman** [comment]
Possibly related, the indents for nested list items is off.  Consider the following:
```
Markdown
* 1
  * 2
    * 3
      * 4
        * 5
          * 6
            * 7
```
The above renders as fully nested in Markdown.

The following indentation is needed to get it to render correctly in github_flavored_markdown:
```
github_flavored_markdown (indents, indents from the line above)
* 1                             0
  * 2                           2   2
     * 3                        5   3
         * 4                    9   4
             * 5                13  4
                 * 6            17  4
                     * 7        21  4
```
The workaround to use 4 spaces works.
